src/composables/trade/useTrading.ts

```diff
@@ -103,15 +106,30 @@ export default function useTrading(
       return 'balancer';
     }
 
-    return tradeGasless.value && isGnosisSupportedOnNetwork.value
-      ? 'gnosis'
-      : 'balancer';
+    if (tradeGasless.value && isGnosisSupportedOnNetwork.value) {
+      return 'gnosis';
+    } else {
+      return (
+        // Currently Relayer only suitable for ExactIn and non-eth swaps
+        exactIn.value &&
+          tokenInAddressInput.value.toLowerCase() !==
+            AddressZero.toLowerCase() &&
+          tokenOutAddressInput.value.toLowerCase() !==
+            AddressZero.toLowerCase() &&
+          joinExit.swapInfo.value?.returnAmount &&
+          !joinExit.swapInfo.value?.returnAmount.isZero()
+          ? 'joinExit'
+          : 'balancer'
```

Comment:
This is very difficult to parse, please refactor into more understandable conditional variables.

Reply:
👍